projects/plugins/jetpack/extensions/shared/components/connect-banner/index.tsx

```diff
@@ -17,25 +19,52 @@ interface ConnectBannerProps {
 
 import './style.scss';
 
+const getRedirectUri = () => {
+	const pathname = window?.location?.pathname.replace( 'wp-admin/', '' );
+	const search = window?.location?.search;
+
+	return `${ pathname }${ search }`;
+};
+
 const ConnectBanner: FC< ConnectBannerProps > = ( { block, explanation = null } ) => {
-	const checkoutUrl = `${ window?.Jetpack_Editor_Initial_State?.adminUrl }admin.php?page=my-jetpack#/connection`;
-	const { autosaveAndRedirect, isRedirecting } = useAutosaveAndRedirect( checkoutUrl );
+	const [ isRedirecting, setIsRedirecting ] = useState( false );
+	const [ isSaving, setIsSaving ] = useState( false );
+	const { handleConnectUser } = useConnection( {
+		from: 'editor',
+		redirectUri: getRedirectUri(),
+		autoTrigger: false,
+		skipPricingPage: true,
+	} );
+	const { autosave } = useAutosaveAndRedirect();
 	const { tracks } = useAnalytics();
 
 	const goToCheckoutPage = ( event: MouseEvent< HTMLButtonElement > ) => {
-		tracks.recordEvent( 'jetpack_editor_connect_banner_click', { block } );
-		autosaveAndRedirect( event );
+		setIsSaving( true );
+		autosave( event ).then( () => {
+			tracks.recordEvent( 'jetpack_editor_connect_banner_click', { block } );
+			setIsRedirecting( true );
+			setIsSaving( false );
+		} );
 	};
 
+	useEffect( () => {
+		// The redirection is handled this way to ensure we get the right redirectUri
+		// In the case that the post is new and unsaved, the component requires a re-render
+		// in order to get the correct URI when the handler is called.
+		if ( isRedirecting ) {
```

Comment:
Another nitpick, this variable name could be something like `isReadyToRedirect` or something like that. To me, calling an action (`handleConnectuser`) when `isRedirecting` is `true` makes me think that it'll interrupt the redirect and do the other thing instead.

Reply:
Nice workaround, btw 😅